client/js/pages/archive/index.js

```diff
@@ -9,7 +9,24 @@ const prettyBytes = require('pretty-bytes')
 const preview = require('../../components/preview')
 
 const archivePage = (state, prev, send) => {
-  // TODO: style the error handling
+  // XXX: have an error enum?
+  if (state.archive.error && state.archive.error.message === 'Invalid key') {
+    // TODO: update below HTML
+    return html`
+      <div>
+        ${header(state, prev, send)}
+        <div class="error-page">
+          <div class="mb3">
+            <img src="./public/img/dat-data-logo.svg" />
+          </div>
+          <h3>404 - Sorry, no dat here.</h3>
+          <p class="mb3">We couldn't find the droids you were looking for. Is the link correct?</p>
+          <p class="mb4">
+            <button class="btn btn--large btn--green take-me-home" href="/">Take me home</button>
+          </p>
+        </div>
+      </div>`
+  }
```

Comment:
seems like this block ^^ should be abstracted away so it can be shared with non-archive routes?